embulk-core/src/main/java/org/embulk/spi/time/TimestampParser.java

```diff
@@ -62,15 +73,19 @@
         public Optional<String> getDate();
     }
 
-    private final JRubyTimeParserHelper helper;
     private final DateTimeZone defaultTimeZone;
+    private final String format;
+    private final RubyDateParser parser;
+    private final Calendar cal;
+    private final List<StrftimeToken> compiledPattern;
 
     @Deprecated
     public TimestampParser(String format, ParserTask task)
     {
         this(task.getJRuby(), format, task.getDefaultTimeZone());
     }
 
+    @VisibleForTesting
```

Comment:
I don't recommend this kind of usage of `@VisibleForTesting`.

If this is used by some main code, we don't need this.

If visibility is needed only for testing, I'd suggest to make a `static` creator method whose name clearly describes it is for testing. For example, `@VisibleForTesting TImestampParser createForTesting(...)`.